tests/e2e/specs/themes/themes.test.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import path from 'path';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	visitAdminPage,
+	installTheme,
+	deleteTheme,
+	isThemeInstalled,
+	__experimentalRest as rest,
+} from "@wordpress/e2e-test-utils";
+
+const themeSlug = "hello-elementor";
+const themeName = "Hello Elementor";
+const themeVersion = '2.0.0';
+const oldThemePath = path.join(
+	__dirname,
+	'..',
+	'..',
+	'test-data',
+	'themes',
+	`${themeSlug}.${themeVersion}.zip`
+);
+
+async function getInstalledThemes() {
+	const themes = await rest({
+		method: 'GET',
+		path: "/wp/v2/themes",
+	});
+	return themes;
+}
+
+async function checkThemeVersion() {
+	const themes = await getInstalledThemes();
+	const theme = themes.find(theme => theme.stylesheet === themeSlug);
+	return theme.version;
+}
+
+async function checkAndInstallTheme() {
+	if (! await isThemeInstalled(themeSlug)) {
+		await installTheme(themeSlug, themeName);
+	} else {
+		await deleteTheme(themeSlug);
+		await installTheme(themeSlug, themeName);
+	}
+}
+
+async function uploadOldTheme() {
+	await visitAdminPage('theme-install.php');
+	await page.waitForSelector('.upload-view-toggle');
+	await page.click('.upload-view-toggle');
+	const input = await page.$('#themezip');
+	await input.uploadFile(oldThemePath);
+	await page.click('#install-theme-submit');
+
+}
+
+describe('Manage uploading new theme versions', () => {
+
+	it('should replace a theme when uploading a new version', async () => {
+		await checkAndInstallTheme();
+		await uploadOldTheme();
+
+		const upgradeHeading = await page.waitForSelector('.update-from-upload-heading');
+		expect(
+			await upgradeHeading.evaluate((element) => element.textContent)
+		).toContain('This theme is already installed.');
+
+		await page.click('a.update-from-upload-overwrite');
+
+		await page.waitForSelector('.wrap');
+		const updatingParagraphs = await page.$$('.wrap p');
+		let mergedMessages = await Promise.all(
+			updatingParagraphs.map((message) => message.evaluate((element) => element.textContent))
+		);
+		mergedMessages = mergedMessages.join(' ');
```

Comment:
It's not really clear to me what these do. Couldn't we just do something like below?

```js
const messageWrapper = await page.$('.wrap');
const messages = await messageWrapper.evaluate(element => element.textContent);
```